src/lua/functions/creatures/creature_functions.cpp

```diff
@@ -1186,3 +1191,80 @@ int CreatureFunctions::luaCreatureClearIcons(lua_State* L) {
 	Lua::pushBoolean(L, true);
 	return 1;
 }
+
+int CreatureFunctions::luaCreatureAttachEffectById(lua_State* L) {
+	// creature:attachEffectById(effectId, [temporary])
+	const auto &creature = Lua::getUserdataShared<Creature>(L, 1);
+	if (!creature) {
+		Lua::reportErrorFunc(Lua::getErrorDesc(LUA_ERROR_CREATURE_NOT_FOUND));
+		Lua::pushBoolean(L, false);
```

Comment:
You just need to use reportErrorFunc, it already pushes the string with the error, so pushing bool or nil is redundant. (Same for the others)